src/client/content_mapblock.cpp

```diff
@@ -445,7 +475,15 @@ void MapblockMeshGenerator::drawSolidNode()
 	f32 texture_coord_buf[24];
 	box.MinEdge += cur_node.origin;
 	box.MaxEdge += cur_node.origin;
+	if (cur_node.f->drawtype == NDT_LIQUID) {
+		cur_liquid.y_offset_allow_wave =
+				enable_waving_water && cur_node.f->waving == 3 ? -0.002f : 0.0f;
```

Comment:
maybe a dumb question but can't we steal a bit from the color? or do we use it?

Reply:
Colors are just u8 in the vertex buffer, so I'm not sure.
The normal would probably be easier to abuse, e.g. by setting normal.X > 2 or so if it should wave, and subtracting 2 in the shader.

I've used the pos thing because it came naturally with the flowing liquid thing I had before.